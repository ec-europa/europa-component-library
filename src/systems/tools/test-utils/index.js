const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
