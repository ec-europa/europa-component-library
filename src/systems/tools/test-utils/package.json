{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
