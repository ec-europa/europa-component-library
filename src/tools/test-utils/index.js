const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
