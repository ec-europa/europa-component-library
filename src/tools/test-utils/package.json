{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
