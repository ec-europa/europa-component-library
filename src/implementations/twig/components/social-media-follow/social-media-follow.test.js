/**
 * @jest-environment jsdom
 */

import { merge, renderTwigFileAsNode } from '@ecl/test-utils';
import demoData from '@ecl/specs-component-social-media-follow/demo/data';

describe('Social Media Follow', () => {
  const template = '@ecl/social-media-follow/social-media-follow.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);
  describe('Default', () => {
    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(demoData, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(demoData, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Horizontal', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(demoData)).resolves.toMatchSnapshot();
    });
  });

  describe('Vertical', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(
        render({
          ...demoData,
          variant: 'vertical',
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
