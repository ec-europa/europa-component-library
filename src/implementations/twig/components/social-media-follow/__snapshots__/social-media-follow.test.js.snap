// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Media Follow Default renders correctly with extra attributes 1`] = `
<jest>
  <div
    class="ecl-social-media-follow"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <p
      class="ecl-social-media-follow__description"
    >
      Follow the latest progress and learn more about getting involved.
    </p>
    <ul
      class="ecl-social-media-follow__list"
    >
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#twitter"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#twitter_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#facebook"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#facebook_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#instagram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#instagram_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Instagram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#linkedin"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#linkedin_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#mastodon"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#mastodon_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Mastodon
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram2"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram2_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram2
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-social-media-follow__link"
          href="/example"
        >
          Other social networks
        </a>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`Social Media Follow Default renders correctly with extra class names 1`] = `
<jest>
  <div
    class="ecl-social-media-follow custom-class custom-class--test"
  >
    <p
      class="ecl-social-media-follow__description"
    >
      Follow the latest progress and learn more about getting involved.
    </p>
    <ul
      class="ecl-social-media-follow__list"
    >
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#twitter"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#twitter_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#facebook"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#facebook_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#instagram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#instagram_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Instagram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#linkedin"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#linkedin_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#mastodon"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#mastodon_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Mastodon
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram2"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram2_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram2
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-social-media-follow__link"
          href="/example"
        >
          Other social networks
        </a>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`Social Media Follow Horizontal renders correctly 1`] = `
<jest>
  <div
    class="ecl-social-media-follow"
  >
    <p
      class="ecl-social-media-follow__description"
    >
      Follow the latest progress and learn more about getting involved.
    </p>
    <ul
      class="ecl-social-media-follow__list"
    >
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#twitter"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#twitter_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#facebook"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#facebook_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#instagram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#instagram_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Instagram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#linkedin"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#linkedin_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#mastodon"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#mastodon_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Mastodon
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram2"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram2_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram2
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-social-media-follow__link"
          href="/example"
        >
          Other social networks
        </a>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`Social Media Follow Vertical renders correctly 1`] = `
<jest>
  <div
    class="ecl-social-media-follow ecl-social-media-follow--vertical"
  >
    <p
      class="ecl-social-media-follow__description"
    >
      Follow the latest progress and learn more about getting involved.
    </p>
    <ul
      class="ecl-social-media-follow__list"
    >
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#twitter"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#twitter_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#facebook"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#facebook_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#instagram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#instagram_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Instagram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#linkedin"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#linkedin_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#mastodon"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#mastodon_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Mastodon
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-follow__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram2"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-follow__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="/icon-social.svg#telegram2_hover"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram2
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-follow__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-social-media-follow__link"
          href="/example"
        >
          Other social networks
        </a>
      </li>
    </ul>
  </div>
</jest>
`;
